### PR TITLE
[feature] Unused avatar and header cleanup

### PR DESCRIPTION
**Pull request by tsmethurst**
This PR adds logic for cleaning up unused avatars and headers (ie., avatars and headers that have been replaced by a profile update, but are still sitting around in storage).

The new functionality looks like this:

1. Select all headers + avatars (in batches on 20).
2. For each avatar + header, check if it's currently in use.
3. If it's not, delete it from storage, then from the database.

This has been added to the processor's MediaRemotePrune function, which has been renamed to MediaPrune to reflect that it's not just remote media being pruned anymore.

This cleanup will be run on instance startup, and every night @ midnight along with the remote media cache prune job.

My tests on goblin.technology and gts.superseriousbusiness.org showed a saving of 1gb-2gb after running this cleanup :)

closes https://github.com/superseriousbusiness/gotosocial/issues/535

### CODE REVIEW

File: internal/media/manager.go [hunk 1]
```diff
@@ -224,5 +189,66 @@ func (m *manager) Stop() error {
...
 	return cronErr
 }
+
+func scheduleCleanupJobs(m *manager) error {
+	// create a new cron instance for scheduling cleanup jobs
+	c := cron.New(cron.WithLogger(&logrusWrapper{}))
+	cancels := []context.CancelFunc{}
+
+	pruneMetaCtx, pruneMetaCancel := context.WithCancel(context.Background())
+	cancels = append(cancels, pruneMetaCancel)
```
Comment (NyaaaWhatsUpDoc) [inline internal/media/manager.go:202]: if you rename pruneMetaCtx to just pruneCtx you can reuse this cancellable context for both set of the pruning jobs :)

**Comment by NyaaaWhatsUpDoc** [comment]
Aside from single comment this looks great :+1:

File: internal/media/manager.go (hunk 1 above)
Comment (tsmethurst) [reply internal/media/manager.go]: Ack! Updated it :)

**Comment by igalic** [comment]
question for @NyaaaWhatsUpDoc, now that my brain has booted  (17:35 local time)

> if you rename pruneMetaCtx to just pruneCtx you can reuse this cancellable context for both set of the pruning jobs :)

won't reusing a variable in a different context make it harder for the GC to know if or when to clean it up?

**Pull request merged**